### ISSUE TITLE
[Request] Support non-Java users for whitelist

### ISSUE THREAD

**Issue by kaysond** [itzg/mc-image-helper#448]
I want to use env vars in `docker-minecraft-server` to manage my whitelist, but it makes the container fail to start up because mc-image-helper chokes on the usernames and UUIDs provided by floodgate. It looks like the API endpoint is hardcoded to minecraft here:

https://github.com/itzg/mc-image-helper/blob/a0b85215944f72cae65ab8d94dea2e01dfe781a1/src/main/java/me/itzg/helpers/users/PlayerdbUserApi.java#L26

playerdb does support xbox and steam accounts via dedicated endpoints, though there seems to be some conversion needed for the floodgate uuid (see https://wiki.geysermc.org/floodgate/faq/#obtaining-uuids-for-floodgate-players)

It would probably need some kind of marker in the env var to indicate a different account type, and maybe there's a reason this can't be supported, but it would be nice to have!

**Comment by itzg** [comment]
Thanks, I had only used/tested with Java usernames of players connected to Xbox user. 

The feature gap wasn't intended. Yes, perhaps a prefix like "xbox:" or "floodgate:" on the username could trigger the new lookup?

**Comment by kaysond** [comment]
> Thanks, I had only used/tested with Java usernames of players connected to Xbox user.
> 
> The feature gap wasn't intended. Yes, perhaps a prefix like "xbox:" or "floodgate:" on the username could trigger the new lookup?

Yeah I was thinking something like that. The env var makes things way easier than having to manually lookup/convert the floodgate names to UUIDs!

FYI - one complication is that it seems you have to specify the user's name with the floodgate prefix. It defaults to `.` but can be changed in the config:

![image](https://github.com/itzg/mc-image-helper/assets/1147328/d6f0690b-a9c8-410b-9d1b-fc5d47eb951d)
Source: https://wiki.geysermc.org/geyser/faq/

**Comment by kaysond** [comment]
There also seems to be some conversion of spaces in a gamertag, for example, to underscores. But in playerdb, it returns the name without spaces (even if you send the request with a space)
